File: app/src/main/java/com/litus_animae/refitted/data/ExerciseRepository.java
package com.litus_animae.refitted.data;

import androidx.lifecycle.LiveData;

import com.litus_animae.refitted.models.ExerciseRecord;
import com.litus_animae.refitted.models.ExerciseSet;
import com.litus_animae.refitted.models.SetRecord;

import java.util.List;

public interface ExerciseRepository {
    void loadExercises(String day, String workoutId);

    void storeSetRecord(SetRecord record);

    LiveData<List<ExerciseSet>> getExercises();

    LiveData<List<ExerciseRecord>> getRecords();

    void shutdown();
}

File: app/src/main/java/com/litus_animae/refitted/data/ExerciseRepository.kt
package com.litus_animae.refitted.data

import com.litus_animae.refitted.data.room.ExerciseDao
import com.litus_animae.refitted.models.ExerciseRecord
import com.litus_animae.refitted.models.ExerciseSet
import com.litus_animae.refitted.models.SetRecord
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.StateFlow

interface ExerciseRepository {
    suspend fun loadExercises(day: String, workoutId: String)
    suspend fun storeSetRecord(record: SetRecord)
    fun loadWorkoutRecords(workoutId: String)
    val exercises: Flow<List<ExerciseSet>>
    val exercisesAreLoading: StateFlow<Boolean>
    val records: Flow<List<ExerciseRecord>>
    val workoutRecords: Flow<List<ExerciseDao.ExerciseCompletionRecord>>
}